### ISSUE TITLE
Cloudlinux 8 repo

### ISSUE THREAD

**Issue by FlokiNET** [engintron/engintron#1385]
Engintron uses wrong sources on cloudlinux 8 (same error as with epel) and therefore shows "Welcome to Nginx on Cloudlinux" showing on all sites after upgrading Engintron on cPanel.

**Comment by CarlosLopezES** [comment]
Hi!

I had the same problem, removed default Nginx folder (rm -rf /usr/share/nginx/html) and it's done.

**Comment by FlokiNET** [comment]
Yes you can create this way temporary fix but still setups a wrong Nginx version as the Cloudlinux / Almalinux source is 1.14 while Nginx.com is 1.21

**Comment by fevangelou** [comment]
Under /etc/yum.repos.d/ locate the CloudLinux (main) repo and below each line that reads `enabled=1` add this line `exclude=nginx*`.

The do a `dnf clean all` and re-install Nginx.

If you spot the correct repo, please paste it back here so I can add an exclusion like I do with EPEL.

**Comment by fevangelou** [comment]
For the record, on "plain" Almalinux there is no such issue, so it's most likely the CloudLinux repo...

**Comment by CarlosLopezES** [comment]
Hi,

The actual repo is "cloudlinux-x86_64-server-8" and "appstream", but I think that CloudLinux repos are constantly being modified.

Thanks!

**Comment by fevangelou** [comment]
Thank you @CarlosLopezES

It'd be interesting to hear back from @FlokiNET in any case...

**Comment by FlokiNET** [comment]
Yes and the mentioned repo "cloudlinux-x86_64-server-8" is not existing as file so cant be edited.

The reply we got from Cloudlinux:
We've figured out that nginx module in CloudLinux is enabled by default as a dependency. It's possible to disable this module by using dnf module disable nginx
but this potentially can be dangerous and could break PHP on your server so I can't recommend you to do this.

 it's safer to use a workaround in this case:
# yum install nginx --disablerepo=cloudlinux-x86_64-server-8

**Comment by Friends4U** [comment]
That only installs nginx but does not work (in my case), it broke Engintron.
Is there a better solution for this

**Comment by manoaratefy** [comment]
The best solution is to put nginx.org repository at a higher priority than the dynamic repository of CloudLinux (cloudlinux-x86_64-server-8) as we can't exclude a package from this due to the lack of the configuration file.

Here is a topic on CloudLinux forum about this: https://forum.cloudlinux.com/forum/cloudlinux-os-features/cloudlinux-os-kernel-and-system-packages/38739-how-to-exclude-a-package-from-cloudlinux-dynamic-repository#post38740

**Comment by Friends4U** [comment]
I did it, but it does not seem to do anything...
Oh well... I'll leave it like this and hope it gets fixed in an update.

**Comment by gu1113** [comment]
the link to the cloudlinux forum didnt work for me either

it did removing the nginx default folder;  /usr/share/nginx/html

Could be any problem just to keep using the nginx version providing by CL ?

**Comment by ajquick** [comment]
Thank you for providing a way to get it running. I installed it this morning and panicked when I saw all the websites on my server replaced with "Welcome to Nginx". Glad to know all I need to do is delete that folder.

Hopefully the issues with Cloudlinux and the outdated repos is fixed soon.

**Comment by FlokiNET** [comment]
Just as an update for people searching for this issue:
there is no solution for it so far, the only option is to move away from engintron and use the build in nginx option from cpanel (which we did)

**Comment by fevangelou** [comment]
Of course there's a solution. I mentioned it already.

What you mean though is that CloudLinux somehow scared you that Nginx will not work correctly if it comes from the official repos (whaaaa?)...

As you wrote earlier from their response:

> [...]but this potentially can be dangerous and could break PHP on your server so I can't recommend you to do this.

Seriously? Unless they are ignorant (which I doubt) this seems more like gatekeeping.

**Comment by manoaratefy** [comment]
> Just as an update for people searching for this issue: there is no solution for it so far, the only option is to move away from engintron and use the build in nginx option from cpanel (which we did)

Just look at my answer here: https://github.com/engintron/engintron/issues/1385#issuecomment-1149613278
You have just to change the repositories priority.

**Comment by fevangelou** [comment]
Changing the priority is one way. The best is to exclude Nginx though, as this might be needed not just in CloudLinux but in some other repo. Engintron already "corrects" Nginx exclusion for EPEL for example...

**Comment by mofolo** [comment]
I had to completely reboot the server plus delete the home directory for this to work again..
Setting the priority didn't seem to work and Engintron still pulls from @System `cloudlinux-x86_64-server-8`
